nomos-services/cryptarchia-consensus/src/leadership.rs

```diff
@@ -1,20 +1,38 @@
-use cl::{input::InputWitness, note::NoteWitness, nullifier::Nullifier};
+use cl::{
+    note::NoteWitness,
+    nullifier::{Nullifier, NullifierSecret},
+    InputWitness,
+};
 use cryptarchia_engine::Slot;
 use leader_proof_statements::{LeaderPrivate, LeaderPublic};
 use nomos_core::header::HeaderId;
 use nomos_ledger::{leader_proof::Risc0LeaderProof, Config, EpochState, NoteTree};
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
 pub struct Leader {
     // for each block, the indexes in the note tree of the notes we control
-    notes: HashMap<HeaderId, Vec<InputWitness>>,
+    notes: HashMap<HeaderId, Vec<NoteWitness>>,
+    nf_sk: NullifierSecret,
```

Comment:
The assumption here is that the leader should use the same sk for all notes? is this better than the old implementation that supported different sk's?

Reply:
> The assumption here is that the leader should use the same sk for all notes?
Yes
> is this better than the old implementation that supported different sk's?
No, but proper support would require wallet-facing api/structures which we haven't built yet